robocorp-code/src/robocorp_code/inspector/image/_image_inspector.py

```diff
@@ -0,0 +1,145 @@
+import threading
+import typing
+from typing import Any, Callable, Optional, List
+
+from robocorp_ls_core.protocols import IEndPoint
+from robocorp_ls_core.callbacks import Callback
+from robocorp_ls_core.robotframework_log import get_logger
+
+from ._bridge import ImageBridge
+
+log = get_logger(__name__)
+
+
+class _PickerThread(threading.Thread):
+    def __init__(
+        self,
+        bridge: Optional[ImageBridge],
+        confidence: Optional[int],
+        on_pick,
+    ) -> None:
+        threading.Thread.__init__(self)
+
+        self._stop_event = threading.Event()
+        self.bridge = bridge
+        self.on_pick = on_pick
+        self.confidence = confidence
+
+    def run(self) -> None:
+        try:
+            self._run()
+        except Exception as e:
+            import traceback
+
+            traceback.print_exc()
+            log.exception("Exception occurred:", e)
+
+    def _run(self) -> None:
+        assert self.bridge is not None
+
+        result_dict = self.bridge.pick(confidence=self.confidence)
+        self.on_pick(result_dict)
+
+
+class _ValidateThread(threading.Thread):
+    def __init__(
+        self,
+        bridge: Optional[ImageBridge],
+        image_base64: str,
+        confidence: Optional[int],
+        on_validate,
+    ) -> None:
+        threading.Thread.__init__(self)
+
+        self._stop_event = threading.Event()
+        self.bridge = bridge
+        self.image_base64 = image_base64
+        self.confidence = confidence
+        self.on_validate = on_validate
+
+    def run(self) -> None:
+        try:
+            self._run()
+        except Exception:
+            import traceback
+
+            traceback.print_exc()
```

Comment:
the `import traceback`/`traceback.print_exc()` should be removed and `log.exception()` should be used here.